### PR TITLE
add rosdep package libadolc-dev

### PR DESCRIPTION
**Pull request by yakunouyang**
Please add the following dependency to the rosdep database.

## Package Upstream Source:

https://github.com/coin-or/adol-c

## Purpose of using this:

This dependency is being used for automatic differentiation along with [IPOPT](https://github.com/coin-or/Ipopt) in optimal control applications.

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bullseye/libadolc-dev
- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/search?searchon=names&keywords=libadolc-dev
- Fedora: https://packages.fedoraproject.org/
  - not available
- Arch: https://www.archlinux.org/packages/
  - https://aur.archlinux.org/packages/adol-c
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/sci-libs/adolc
- macOS: https://formulae.brew.sh/
  - https://github.com/brewsci/homebrew-science/blob/master/Formula/adol-c.rb
- Alpine: https://pkgs.alpinelinux.org/packages
  - not available
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?channel=22.11&show=adolc&from=0&size=50&sort=relevance&type=packages&query=adolc
- openSUSE: https://software.opensuse.org/package/
  - https://software.opensuse.org/package/adolc

### CODE REVIEW

**Review by sloretz — APPROVED** [review]

**Comment by yakunouyang** [comment]
Hi, just want to check in and see if there are any updates @sloretz

File: rosdep/base.yaml [hunk 1]
```diff
@@ -2401,6 +2401,11 @@ libabsl-dev:
     '*': [libabsl-dev]
     bionic: null
     focal: null
+libadolc-dev:
+  debain: [libadolc-dev]
```
Comment (emersonknapp) [inline rosdep/base.yaml:2405]:
```suggestion
  debian: [libadolc-dev]
```

File: rosdep/base.yaml [hunk 2]
```diff
@@ -2401,6 +2401,11 @@ libabsl-dev:
...
+  gentoo: [sci-libs/adolc]
+  nixos: [adolc]
```
Comment (emersonknapp) [inline rosdep/base.yaml:2407]:
```suggestion
  nixos: [adolc]
  opensuse: [adolc-devel]
```

**Review by emersonknapp — CHANGES_REQUESTED** [review]
Sorry for the slow turnaround there. I noticed that there's a typo `debain` which should be `debian`. I've left the suggestion inline, and an `opensuse` addition which is optional but recommended. If that's all right, go ahead and commit them, then we can move forward.

**Comment by yakunouyang** [comment]
fixed, sorry for the typo :smiling_face_with_tear:

**Review by emersonknapp — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**